components/brave_wallet/common/solana_utils.cc

```diff
@@ -27,15 +28,19 @@ void CompactU16Encode(uint16_t u16, std::vector<uint8_t>* compact_u16) {
   }
 }
 
-bool Base58Decode(const std::string& str,
-                  std::vector<uint8_t>* ret,
-                  int max_ret_len) {
+bool Base58Decode(const std::string& str, std::vector<uint8_t>* ret, int len) {
   DCHECK(ret);
-  return DecodeBase58(str, *ret, max_ret_len);
+  ret->clear();
+  return DecodeBase58(str, *ret, len) && static_cast<int>(ret->size()) == len;
```

Comment:
AFAIK our current usage in the code base want to specify the exact length instead of max length.